### ISSUE TITLE
Cmd('git') failed due to: exit code(129) when updating node

### ISSUE THREAD

**Issue by irrikrlla** [Comfy-Org/ComfyUI-Manager#921]
![屏幕截图 2024-08-01 224219](https://github.com/user-attachments/assets/0a453e5b-ed7d-46d2-b05c-3ef99e5ebe42)
I tried to update a node, with the newest ComfyUImanager, but failed with the e Cmd('git') failed due to: exit code(129). I also tried to fix by modifying the code, but had no idea where is the bug. Is there any idea to solve this? It's the first time I meet the issue with manager.

**Comment by ltdrdata** [comment]
Please show the results of executing the following commands in cmd:

```
git diff --cached
git --version
```

**Comment by irrikrlla** [comment]
![屏幕截图 2024-08-02 083055](https://github.com/user-attachments/assets/b25127bd-afb7-4380-a5c7-5a8e144d3ad6)
Here is the result~
git version is the latest, git diff --cached command gives nothing

**Comment by cheezmil** [comment]
+1
![image](https://github.com/user-attachments/assets/6c0e4c24-ab9c-48f3-8b58-d57d64739d4e)

**Comment by cheezmil** [comment]
![image](https://github.com/user-attachments/assets/bf3376bd-3734-488b-a3df-55c72c2b06dc)
After entering this command, I can click on "try updata" in Comfyui. It's really strange. Do I have to enter this command every time I update? It's too troublesome. What should I do.

**Comment by ltdrdata** [comment]
> After entering this command, I can click on "try updata" in Comfyui. It's really strange. Do I have to enter this command every time I update? It's too troublesome. What should I do.

That directory isn't on a network drive, is it?

**Comment by cheezmil** [comment]
> > After entering this command, I can click on "try updata" in Comfyui. It's really strange. Do I have to enter this command every time I update? It's too troublesome. What should I do.
> 
> That directory isn't on a network drive, is it?

“That directory isn't on a network drive, is it?” No, it may not have anything to do with this.

I found that running cmd as an administrator can solve this problem, but I don't know if it's safe to do so..

**Comment by ltdrdata** [comment]
> > > After entering this command, I can click on "try updata" in Comfyui. It's really strange. Do I have to enter this command every time I update? It's too troublesome. What should I do.
> > 
> > 
> > That directory isn't on a network drive, is it?
> 
> “That directory isn't on a network drive, is it?” No, it may not have anything to do with this.
> 
> I found that running cmd as an administrator can solve this problem, but I don't know if it's safe to do so..

What about permission?

**Comment by cheezmil** [comment]
> > > > After entering this command, I can click on "try updata" in Comfyui. It's really strange. Do I have to enter this command every time I update? It's too troublesome. What should I do.
> > > 
> > > 
> > > That directory isn't on a network drive, is it?
> > 
> > 
> > “That directory isn't on a network drive, is it?” No, it may not have anything to do with this.
> > I found that running cmd as an administrator can solve this problem, but I don't know if it's safe to do so..
> 
> What about permission?

I have found a solution。
https://www.zhihu.com/tardis/bd/art/619858339

Running Comfyui's cmd as an administrator can also solve the problem.

In addition, there is also an issue with ComfyUI Manager. Even though it reports an error with exit code (129), it still shows that the update was successful. The exception should be caught instead of informing the user that the update was successful.

**Comment by n0valis** [comment]
Yes can confirm. ComfyUI does not capture the fail instead claims success.


 
> I have found a solution。 https://www.zhihu.com/tardis/bd/art/619858339
> 

How is that a solutution? That's bloody chinese

**Comment by ltdrdata** [comment]
> Yes can confirm. ComfyUI does not capture the fail instead claims success.
> 
> > I have found a solution。 https://www.zhihu.com/tardis/bd/art/619858339
> 
> How is that a solutution? That's bloody chinese

In Explorer, check the properties of the directory in the permissions section to verify if the owner is set to your account.
